include/deal.II/base/utilities.h

```diff
@@ -578,16 +578,16 @@ namespace Utilities
    * $p_i\in [0,N)$ and $p_i\neq p_j$ for $i\neq j$), produce the reverse
    * permutation $q_i=N-1-p_i$.
    */
-  std::vector<unsigned long long int>
-  reverse_permutation(const std::vector<unsigned long long int> &permutation);
+  std::vector<std::uint64_t>
+  reverse_permutation(const std::vector<std::uint64_t> &permutation);
 
   /**
    * Given a permutation vector (i.e. a vector $p_0\ldots p_{N-1}$ where each
    * $p_i\in [0,N)$ and $p_i\neq p_j$ for $i\neq j$), produce the inverse
    * permutation $q_0\ldots q_{N-1}$ so that $q_{p_i}=p_{q_i}=i$.
    */
-  std::vector<unsigned long long int>
-  invert_permutation(const std::vector<unsigned long long int> &permutation);
```

Comment:
We could also just templatize these two functions instead of repeating the definition for the two number types.

Reply:
Well, we do have this one: https://github.com/dealii/dealii/blob/c8233d7c7984e0cc1ddb1f19abb4295366cce289/source/base/utilities.cc#L944-L980
It seems we should simply move this above the other two functions - let me try that out in a minute; regarding the declaration I like to state the supported types given that it is only two, i.e., I prefer a compile error over a link error.

Reply:
I have now added a second commit that removes the code duplication in favor of the templated function we had in this file already.